v2/internal/genericarmclient/generic_client.go

```diff
@@ -138,12 +139,14 @@ func (client *GenericClient) createOrUpdateByID(
 	resourceType := metrics.GetTypeFromResourceID(resourceID)
 
 	client.metrics.RecordAzureRequestsTime(resourceType, time.Since(requestStartTime), metrics.HttpPut)
-	client.metrics.RecordAzureRequestsTotal(resourceType, resp.StatusCode, metrics.HttpPut)
 
 	if err != nil {
 		client.metrics.RecordAzureFailedRequestsTotal(resourceType, metrics.HttpPut)
 		return nil, err
 	}
+
+	client.metrics.RecordAzureRequestsTotal(resourceType, resp.StatusCode, metrics.HttpPut)
```

Comment:
Might be worth updating the comment as well for the method inside `arm_client_metrics.go`. With this change it will not longer be `total(failed + successful)` and just the successful requests to the ARM server